src/Tempest/Core/src/Kernel/LoadDiscoveryClasses.php

```diff
@@ -95,12 +95,10 @@ public function __invoke(): void
                         }
                     }
 
-                    if ($input instanceof ClassReflector && $input->hasAttribute(DoNotDiscover::class)) {
-                        continue;
-                    }
-
                     if ($input instanceof ClassReflector) {
-                        $discovery->discover($input);
+                        if ($this->shouldDiscover($discovery, $input)) {
```

Comment:
Not a fan of the nesting here, but it also avoids repeating the condition above 🫠